doctr/models/detection/linknet/pytorch.py

```diff
@@ -174,17 +178,26 @@ def compute_loss(
         seg_target, seg_mask = torch.from_numpy(seg_target).to(dtype=out_map.dtype), torch.from_numpy(seg_mask)
         seg_target, seg_mask = seg_target.to(out_map.device), seg_mask.to(out_map.device)
 
-        # BCE loss
+        # Focal loss
         bce_loss = F.binary_cross_entropy_with_logits(out_map, seg_target, reduction='none')
+        if gamma and gamma < 0:
+            raise ValueError("Value of gamma should be greater than or equal to zero.")
+        # Convert logits to prob, compute gamma factor
+        pred_prob = torch.sigmoid(out_map)[seg_mask]
```

Comment:
Let's avoid doing the computation twice (cf. `prob_map` of the dice loss, which we could put before so that it can be masked here)